nemo_skills/pipeline/utils/cluster.py

```diff
@@ -197,6 +197,22 @@ def get_env_variables(cluster_config):
             if env_var_name not in _logged_optional_env_vars:
                 LOG.info(f"Optional environment variable {env_var_name} not found in user environment; skipping.")
                 _logged_optional_env_vars.add(env_var_name)
+
+    # replace placeholders with actual env var values from the environment where
+    # the job is being launched, not where its being run.
+    for key, value in env_vars.items():
+        if isinstance(value, str) and "$" in value:
+            if key in os.environ:
+                env_vars[key] = os.path.expandvars(value)
+                LOG.info(
+                    f"Resolved environment variable {key} inside the placeholder value: {value} with {env_vars[key]}"
+                )
+            else:
+                raise ValueError(f"Cannot resolve environment variable {key} inside the placeholder value: {value}")
+
+    # Inject env_vars into os.environ for resolution via os.path.expandvars in other parts of the code
+    os.environ.update(env_vars)
```

Comment:
we shouldn't be doing this - we don't want to change the environment for one job based on values from another job. As in, cluster config can be changed between pipeline launches, but python session might be the same and if you write to os.environ directly it can lead to hard to catch issues when one job affects another job in unexpected ways. Can we instead pass things through env_vars dict?